### PR TITLE
NS-939, initial commit of piazza nessie import and transform jobs

### PR DESCRIPTION
**Pull request by jblowe**
https://jira.ets.berkeley.edu/jira/browse/NS-939

### CODE REVIEW

**Comment by sandeepmjay** [comment]
Thanks for the work, @jblowe. I have done a round of review. @johncrossman @paulkerschen keeping you apprised of changes. Mind giving a once over?

File: nessie/jobs/transform_piazza_api_data.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        app.logger.info(f'Doing Piazza API data transform job... {s3_source} > {s3_dest}')
+        objects = s3.get_keys_with_prefix(s3_source)
+        app.logger.info(f'Will transform {len(objects)} objects from {s3_source} and put results to {s3_dest}.')
+        skip_count = 0
+        new_objects = 0
+        for o in objects:
+            file_name = o.split('/')[-1]
+            app.logger.info(f'processing {file_name}')
+            # file_name is like 'daily_2020-08-14.zip'
+            # datestamp = file_name.split('_')[1].split('.')[0]
```
Comment (paulkerschen) [inline nessie/jobs/transform_piazza_api_data.py:62]: Commented out `datestamp` assignment - not needed?

File: nessie/jobs/transform_piazza_api_data.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                        course_id = subfile.split('/')[-2]
+                        file_type = json_file.split('_')[0]
+                        record = piazza_zip_file.read(subfile)
+                        with tempfile.TemporaryFile() as result:
+                            s3_object = f'{s3_dest}/{file_type}/{course_id}/{json_file}'
+                            if s3.object_exists(s3_object):
+                                skip_count += 1
+                                continue
+                            result.write(record)
+                            # result.write(json.dumps(record).encode('utf-8'))
```
Comment (paulkerschen) [inline nessie/jobs/transform_piazza_api_data.py:77]: And here - commented-out line from development/debugging?

**Comment by jblowe** [comment]
https://jira.ets.berkeley.edu/jira/browse/NS-939

**Comment by paulkerschen** [comment]
Thanks for the work, @jblowe - looks great to me, and appreciate the inline documentation! Just noticed a couple of commented-out lines that might be left over from the development process.

File: nessie/jobs/transform_piazza_api_data.py (hunk 1 above)
Comment (jblowe) [reply nessie/jobs/transform_piazza_api_data.py:62]: Well, Sandeep said I could leave that in there for now, as I am planning to use it shortly. But I'll take it out if you like.

File: nessie/jobs/transform_piazza_api_data.py (hunk 2 above)
Comment (jblowe) [reply nessie/jobs/transform_piazza_api_data.py:77]: Well, Sandeep said I could leave that in there for now, as I am planning to use it shortly. But I could go if you like!

File: nessie/jobs/transform_piazza_api_data.py (hunk 1 above)
Comment (paulkerschen) [reply nessie/jobs/transform_piazza_api_data.py:62]: Ah, if they’re in here for the future and not the past, that’s fine. Thanks!

**Pull request merged**